src/en/darkscience/src/eu/kanade/tachiyomi/extension/en/darkscience/DarkScience.kt

```diff
@@ -0,0 +1,156 @@
+    // backup restore. And in a backup, only `url` and `title` are preserved.
+    override fun fetchMangaDetails(manga: SManga): Observable<SManga> = Observable.just(initTheManga(manga))
+
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val chapters = mutableListOf<SChapter>()
+
+        var archivePage: Document? = client.newCall(GET(archiveUrl, headers)).execute().asJsoup()
+
+        var chLast = 0.0F
+
+        while (archivePage != null) {
+            val nextArchivePageUrl = archivePage.selectFirst("""#nav-below .nav-previous > a""")?.attr("href")
+            val nextArchivePage = if (nextArchivePageUrl != null) {
+                client.newCall(GET(nextArchivePageUrl, headers)).execute().asJsoup()
+            } else { null }
+
+            archivePage.select("""#content article header > h2 > a""").forEach {
+                val chTitle = it.text()
+                val chLink = it.attr("href")
+                val chDateMatch = chapterDateRegex.find(chLink)!!
+                val chNumMatch = chapterNumberRegex.find(chTitle)
+                val chDate = DATE_FMT.parse(chDateMatch.groupValues[1])?.time ?: 0L
+                val chNum = chNumMatch?.groupValues?.getOrNull(1)?.toFloat() ?: (chLast + 0.01F)
+
+                chapters.add(
+                    SChapter.create().apply {
+                        name = chTitle
+                        chapter_number = chNum
+                        date_upload = chDate
+                        setUrlWithoutDomain(chLink)
+                    },
+                )
+
+                // This is a hack to make the app not think there’s missing chapters after
+                // a title page.
+                chLast = chNum
+            }
+
+            archivePage = nextArchivePage
+        }
+
+        return Observable.just(chapters)
+    }
+
+    override fun fetchPageList(chapter: SChapter): Observable<List<Page>> {
+        // It would’ve been cleaner and neater to extract chapter information from the archive feed.
+        // Alas, the website provides no structured way of extracting that information. We’d have
+        // to iterate over all »chapters« so far, picking the ones for which the chapter number
+        // regex fails, *guessing* they’re chapter title pages, and then demote all other chapters
+        // to pages of these. As i don’t see a clean way to cache the »chapter« list we’ve fetched
+        // so far, that whole endeavour seems not worth the effort.
+        // So here it is: Each »chapter« having just 1 page.
+        return Observable.just(listOf(Page(0, chapter.url)))
+    }
+
+    override fun fetchImageUrl(page: Page): Observable<String> {
+        val comicPage = client.newCall(GET(baseUrl + page.url, headers)).execute().asJsoup()
+        val imageUrl = comicPage.selectFirst("article.post img.aligncenter")!!.attr("src")
+        return Observable.just(imageUrl)
+    }
```

Comment:
override imageUrlParse instead

Reply:
Didn’t notice that function, for the extension i learned from didn’t use it, either. Thanks, changed.